fe/fe-core/src/main/java/org/apache/doris/planner/external/ExternalHiveScanProvider.java

```diff
@@ -109,12 +112,22 @@ public List<InputSplit> getSplits(List<Expr> exprs)
 
         Configuration configuration = setConfiguration();
         InputFormat<?, ?> inputFormat = HiveUtil.getInputFormat(configuration, inputFormatName, false);
+        List<InputSplit> result;
         if (!hivePartitions.isEmpty()) {
-            return hivePartitions.parallelStream()
+            result = hivePartitions.parallelStream()
                     .flatMap(x -> getSplitsByPath(inputFormat, configuration, x.getSd().getLocation()).stream())
                     .collect(Collectors.toList());
         } else {
-            return getSplitsByPath(inputFormat, configuration, splitsPath);
+            result = getSplitsByPath(inputFormat, configuration, splitsPath);
+        }
+        Optional<Set<Integer>> prunedBuckets = HiveBucketUtil.getPrunedBuckets(
```

Comment:
Looks like we can merge `getPrunedBuckets` and `getPrunedSplitsByBuckets` into one method.
To make it simpler to use?

Reply:
done